File: packages/web/app/(task)/tasks/_query/taskQuery.ts
import { supabase } from "@/app/(core)/supabase";
import { TaskEntity, TaskStatusEntity } from "../_data-access/taskEntity";

export const taskQuery = {
  /** 引数のIDに一致するタスクを取得する */
  fetchTask: async (id: number) => {

    // データを取得する
    const { data, error } = await supabase.from("tasks")
      .select('*')
      .eq("id", id).single();
  
    // エラーをチェックする
    if (error) throw error;

    // 型をキャストする
    const task = data as TaskEntity
  
    return task
  },

  /** タスクステータスを取得する */
  fetchTaskStatuses: async () => {

    // データを取得する
    const { data, error } = await supabase.from("task_statuses")
      .select('*')

    // エラーをチェックする
    if (error) throw error;

    // 型をキャストする
    const taskStatuses = data as TaskStatusEntity[]

    return taskStatuses ?? [];
  }
}

File: packages/ai/llm_demo/task/query/task_query.py
from llm_demo.core.supabase import supabase
from llm_demo.task.model.task import Task

def fetch_tasks():
    '''全タスクを取得する'''
    response = (
        supabase.table("tasks")
        .select("*")
        .execute()
    )
    # タスクモデルに変換して返す
    return [Task(**task_dict) for task_dict in response.data] # type: ignore

# 動作確認用
if __name__ == "__main__":
    tasks = fetch_tasks()
    [print(task) for task in tasks]
